OSDOCS-12595-re: Documented the 4.17.4 z-stream release notes (#84883)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

Issued: 13 November 2024

{product-title} release {product-version}.4, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:8981[RHSA-2024:8981] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:8984[RHSA-2024:8984] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.17.4 --pullspecs
----

[id="ocp-4-17-4-enhancements_{context}"]
==== Enhancements

[id="ocp-4-17-4-enhancements-nodes-pods-short-term-auth_{context}"]
===== Authenticating customer workloads with {gcp-wid-short}

With this release, applications in customer workloads on {product-title} clusters that use {gcp-wid-first} can authenticate by using {gcp-wid-short}.

To use this authentication method with your applications, you must complete configuration steps on the cloud provider console and your {product-title} cluster.

> //For more information, see xref:../nodes/pods/nodes-pods-short-term-auth.adoc#nodes-pods-short-term-auth-configuring-gcp_nodes-pods-short-term-auth[Configuring {gcp-wid-short} authentication for applications on {gcp-short}].

Reviewer: Should this be commented out?

Author: Yes, sorry. https://github.com/openshift/openshift-docs/pull/83682 will comment this link in.

[id="ocp-4-17-4-enhancements-ansible-operator_{context}"]
===== ansible-operator upstream version information

* The `ansible-operator` version now shows the corresponding upstream version information. (link:https://issues.redhat.com/browse/OCPBUGS-43836[*OCPBUGS-43836*])

[id="ocp-4-17-4-bug-fixes_{context}"]
==== Bug fixes

> * Previously, when installing a cluster on an {ibm-title} platform and you wanted to add an existing VPC to the cluster, the {cap-ibm-first} would not adds ports 443, 5000, and 6443 to the security group of the VPC. This situation prevented the VPC from being added to the cluster. With this release, a fix ensures that the {cap-ibm-first} adds the ports to the security group of the VPC so that the VPC gets added to yoru cluster. (link:https://issues.redhat.com/browse/OCPBUGS-44226[*OCPBUGS-44226*])

Reviewer: ```suggestion
* Previously, when installing a cluster on an {ibm-title} platform and adding an existing VPC to the cluster, the {cap-ibm-first} would not add ports 443, 5000, and 6443 to the security group of the VPC. This situation prevented the VPC from being added to the cluster. With this release, a fix ensures that the {cap-ibm-first} adds the ports to the security group of the VPC so that the VPC gets added to your cluster. (link:https://issues.redhat.com/browse/OCPBUGS-44226[*OCPBUGS-44226*])
```

+
Updating these fields did not trigger an update to the control plane machines, and these fields were cleared when the control plane machine set was deleted. With this release, the installation program is updated to no longer populate values that are included in the failure domain configuration. If these values are not defined in a failure domain configuration, for instance on a cluster that is updated to {product-title} {product-version} from an earlier version, the values defined by the installation program are used. (link:https://issues.redhat.com/browse/OCPBUGS-44047[*OCPBUGS-44047*])

* Previously, enabling ESP hardware offload using IPSec on attached interfaces in Open vSwitch broke connectivity due to a bug in Open vSwitch. With this release, OpenShift automatically disables ESP hardware offload on the Open vSwitch attached interfaces, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-44043[*OCPBUGS-44043*])

> * Previously, enabling ESP hardware offload using IPSec on attached interfaces in Open vSwitch broke connectivity due to a bug in Open vSwitch. With this release, OpenShift automatically disables ESP hardware offload on the Open vSwitch attached interfaces, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-43917[*OCPBUGS-43917*])

Reviewer: Since these are the same bug with two tickets, can this be combined into one entry, with two bugs listed at the end? just a thought.

Author: Good catch. OCPBUGS-44043 targets 4.16 and is not attached to the Images advisory so I'll drop this bug.

* Previously, when you installed a private cluster on {gcp-first}, the API firewall rule used the source range of `0.0.0.0/0`. This address allowed non-cluster resources unintended access to the private cluster. With this release, the API firewall rule now only allows resources that have source ranges in the Machine Network to access the private cluster. (link:https://issues.redhat.com/browse/OCPBUGS-43786[*OCPBUGS-43786*])

* Previously, an invalid or unreachable identity provider (IDP) blocked updates to {hcp}. With this release, the `ValidIDPConfiguration` condition in the `HostedCluster` object now reports any IDP errors so that these errors do not block updates to {hcp}. (link:https://issues.redhat.com/browse/OCPBUGS-43746[*OCPBUGS-43746*])

> * Previously, the `attach`,`oc exec`, and `port-forward` commands received an error if you ran the commands through a proxy. With this release, a patch applied to kubectl ensures kubectl can handle any proxy errors with these commands, so that the commands run as expected. (link:https://issues.redhat.com/browse/OCPBUGS-43696[*OCPBUGS-43696*])

Reviewer: ```suggestion
* Previously, the `attach`, `oc exec`, and `port-forward` commands received an error if you ran the commands through a proxy. With this release, a patch applied to kubectl ensures kubectl can handle any proxy errors with these commands, so that the commands run as expected. (link:https://issues.redhat.com/browse/OCPBUGS-43696[*OCPBUGS-43696*])
```

* Previously, {op-system-base-full} CoreOS templates that were shipped by the  Machine Config Operator (MCO) caused node scaling to fail on {rh-openstack-first}. This issue happened because of an issue with `systemd` and the presence of a legacy boot image from older versions of {product-title}. With this release, a patch fixes the issue with `systemd` and removes the legacy boot image, so that node scaling can continue as expected. (link:https://issues.redhat.com/browse/OCPBUGS-42577[*OCPBUGS-42577*])

> * Previously, a restart of a Cluster Version Operator (CVO) pod while the pod was initializing a syncing operation caused the gaurd of a blocked upgrade request to fail. This meant that the blocked upgrade request was unexpectedly accepted. With this release, a CVO pod postpones reconciliation of requests during initialization, so that the guard of a blocked upgrade requests persists after the CVO pod restarts. (link:https://issues.redhat.com/browse/OCPBUGS-42386[*OCPBUGS-42386*]) 

Reviewer: ```suggestion
* Previously, a restart of a Cluster Version Operator (CVO) pod while the pod was initializing a syncing operation caused the guard of a blocked upgrade request to fail. This meant that the blocked upgrade request was unexpectedly accepted. With this release, a CVO pod postpones reconciliation of requests during initialization, so that the guard of a blocked upgrade requests persists after the CVO pod restarts. (link:https://issues.redhat.com/browse/OCPBUGS-42386[*OCPBUGS-42386*]) 
```
